modules/storage-aws-s3/s3/s3.go

```diff
@@ -148,6 +148,61 @@ func (s *s3) RestoreSnapshot(ctx context.Context, className, snapshotID string)
 	return nil
 }
 
+func (s *s3) GetMetaStatus(ctx context.Context, className, snapshotID string) (string, error) {
+	objectName := makeObjectName(className, snapshotID, "snapshot.json")
+	obj, err := s.client.GetObject(ctx, s.config.BucketName(), objectName, minio.GetObjectOptions{})
+	if err != nil {
+		return "", errors.Wrapf(err, "get file %s", objectName)
+	}
+
+	data, err := io.ReadAll(obj)
+	if err != nil {
+		return "", errors.Wrapf(err, "read file %s", objectName)
+	}
+
+	var snapshot snapshots.Snapshot
+	err = json.Unmarshal(data, &snapshot)
+	if err != nil {
+		return "", errors.Wrapf(err, "unmarshal meta")
+	}
+
+	return string(snapshot.Status), nil
+}
+
+func (s *s3) SetMetaStatus(ctx context.Context, className, snapshotID, status string) error {
+	objectName := makeObjectName(className, snapshotID, "snapshot.json")
+	obj, err := s.client.GetObject(ctx, s.config.BucketName(), objectName, minio.GetObjectOptions{})
+	if err != nil {
+		return errors.Wrapf(err, "get file %s", objectName)
+	}
+
+	data, err := io.ReadAll(obj)
+	if err != nil {
+		return errors.Wrapf(err, "read file %s", objectName)
+	}
+
+	var snapshot snapshots.Snapshot
+	err = json.Unmarshal(data, &snapshot)
+	if err != nil {
+		return errors.Wrapf(err, "unmarshal meta")
+	}
```

Comment:
that part looks similar to GetMetaStatus -> could be extracted to common func?

Reply:
done. was able to apply the extracted func to RestoreSnapshot as well